test/remote/account_status_tests.js

```diff
@@ -97,6 +97,67 @@ TestServer.start(config)
   )
 
   test(
+    'account status by email with existing account',
+    function (t) {
+      var email = server.uniqueEmail()
+      return Client.create(config.publicUrl, email, 'password')
+        .then(
+          function (c) {
+            return c.api.accountStatusByEmail(email)
+          }
+        )
+        .then(
+          function (response) {
+            t.ok(response.exists, 'account exists')
+          }
+        )
+    }
+  )
+
+  test(
+    'account status by email with non-existing account',
+    function (t) {
+      var email = server.uniqueEmail()
+      return Client.create(config.publicUrl, email, 'password')
+        .then(
+          function (c) {
+            var nonExistEmail = server.uniqueEmail()
+            return c.api.accountStatusByEmail(nonExistEmail)
+          }
+        )
+        .then(
+          function (response) {
+            t.ok(!response.exists, 'account does not exist')
+          }
+        )
+    }
+  )
+
+  test(
+    'account status by email with invald email',
+    function (t) {
+      var email = server.uniqueEmail()
+      return Client.create(config.publicUrl, email, 'password')
+        .then(
+          function (c) {
+            var invalidEmail = 'notAnEmail'
+            return c.api.accountStatusByEmail(invalidEmail)
+          }
+        )
+        .then(
+          function () {
+            t.fail('should not have successful request')
+          },
+          function (err) {
+            t.equal(err.code, 400)
+            t.equal(err.errno, 107)
+            t.equal(err.message, 'Invalid parameter in request body')
+          }
+        )
+    }
+  )
+
+  test(
```

Comment:
Nice test coverage.  I'm not sure how hard it would be to mock to mock out the relevant bits, but it would also be good to confirm that the customs API functions are being called at the appropriate time, and that we return the correct error when the endpoint gets rate-limited.